client/js/controllers/editor.js

```diff
@@ -1036,6 +1038,130 @@
+
+                $scope.getNodes();
+
+                var prettifyNodetype = function (nodetype) {
+                    var temp = nodetype.split('/');
+                    temp[0] = temp[0].toUpperCase();
+                    if (temp[1]) {
+                        temp[1] = temp[1].charAt(0).toUpperCase() + temp[1].slice(1);
+                        return temp[0] + ' ' + temp[1];
+                    }
+                };
+
+                var autoCompleter = {
+                    getCompletions: function(editor, session, pos, prefix, callback) {
+                        var curLine = session.getDocument().getLine(pos.row);
+                        var curTokens = curLine.slice(0, pos.column).split(/\s+/);
+                        var valTokens = curLine.slice(0, pos.column).split(':');
+                        if (valTokens[0]) {
+                            var nodeval = valTokens[0].lastIndexOf('(');
+                            if (nodeval > -1) {
+                                var valCmd = valTokens[0].slice(nodeval + 1);
+                            }
+                        }
+                        var curCmd = curTokens[0];
+                        if(!curCmd) return;
+                        var node_pos = nodenames.indexOf(curTokens[curTokens.length-2]);
+                        if (curTokens[curTokens.length-2]) {
+                            var brackets = curTokens[curTokens.length-2].split('(');
+                            if (brackets[1]) {
+                                var colon = brackets[1].split(':')[0];
+                                var colon_pos = nodenames.indexOf(colon);
+                            }
+                            var brack_pos = nodenames.indexOf(brackets[brackets.length-1].slice(0, -1));
+                        }
+                        if (valCmd) {
+                            var val_pos = membernodes.indexOf(valCmd);
+                        }
+                        if (node_pos > -1 || brack_pos > -1 || colon_pos > -1 || val_pos > -1) {
+                            var portsList = [];
+                            var dataTypeList = [];
+                            var values = members[val_pos];
+                            if (outports[node_pos]) {
+                                var ports = outports[node_pos];
+                            } else if (outports[brack_pos]) {
+                                var ports = outports[brack_pos];
+                            } else if (outports[colon_pos]) {
+                                var ports = outports[colon_pos];
+                            } else if (inports[node_pos]) {
+                                var ports = inports[node_pos];
+                            } else if (inports[brack_pos]) {
+                                var ports = inports[brack_pos];
+                            } else if (inports[colon_pos]) {
+                                var ports = inports[colon_pos];
+                            }
```

Comment:
You should neve declare the same var over and over again....

Declare it on the top even if it is null and use it here... but don't do this declarations

You also make a switch/case here that can let the block of code more readable... but this is up to you